pkg/otlp/attributes/attributes.go

```diff
@@ -173,16 +160,39 @@ func OriginIDFromAttributes(attrs pcommon.Map) (originID string) {
 	return
 }
 
+// ContainerTagFromResourceAttributes extracts container tags from the given
```

Comment:
IMO it would be nice to document here the `datadog.container.tag` prefix thing as part of the function's documentation.

Another thing that would be relevant to document is what happens in these two edge cases:
1. Two container tags with the same name but different value (`datadog.container.tag.foo`: `bar` and `datadog.container.tag.foo`: `baz`)
2. A container tag that has the same name as a DD container mapping (`datadog.container.tag.container_id`: `bar`, `container.id`: `baz`)

It could be that anything can happen, in any case we should explicitly state it.

Reply:
For 1., I don't think that it is possible to have the same key twice (see [here](https://opentelemetry.io/docs/specs/otel/common/#attribute:~:text=The%20keys%20in%20each%20such%20collection%20are%20unique%2C%20i.e.%20there%20MUST%20NOT%20exist%20more%20than%20one%20key%2Dvalue%20pair%20with%20the%20same%20key.)). If a user sets the same key twice, this will be deduplicated (see [here](https://opentelemetry.io/docs/specs/otel/common/#attribute:~:text=Setting%20an%20attribute%20with%20the%20same%20key%20as%20an%20existing%20attribute%20SHOULD%20overwrite%20the%20existing%20attribute%E2%80%99s%20value)). So I think we can assume that each key is unique here ?

For 2. I think that the semantic convention should take priority, I will document this, add code for this as well as a test case.

For 3., any `pcommon.Value` where `.AsString()` returns an empty string, we will skip adding. I will document this and add a test case.

Reply:
2. and 3. should be addressed in the latest commit

Reply:
> For 1., I don't think that it is possible to have the same key twice (see [here](https://opentelemetry.io/docs/specs/otel/common/#attribute:~:text=The%20keys%20in%20each%20such%20collection%20are%20unique%2C%20i.e.%20there%20MUST%20NOT%20exist%20more%20than%20one%20key%2Dvalue%20pair%20with%20the%20same%20key.)). If a user sets the same key twice, this will be deduplicated (see [here](https://opentelemetry.io/docs/specs/otel/common/#attribute:~:text=Setting%20an%20attribute%20with%20the%20same%20key%20as%20an%20existing%20attribute%20SHOULD%20overwrite%20the%20existing%20attribute%E2%80%99s%20value)). So I think we can assume that each key is unique here ?

You are right :) We can ignore this case. Incidentally, this is another 'feature' allowed in host tags but not here.

> For 2. I think that the semantic convention should take priority, I will document this, add code for this as well as a test case.

:+1:, agreed that that seems like the most reasonable approach.

> For 3., any pcommon.Value where .AsString() returns an empty string, we will skip adding. I will document this and add a test case.

Note that `AsString` will not return an empty string for any type; instead it will convert to a string with some predefined method (e.g. for a slice [it will marshal as JSON](https://github.com/open-telemetry/opentelemetry-collector/blob/2a9d870a57038cb228ef20e7620f18a93c92b9d3/pdata/pcommon/value.go#L398-L399)).

Reply:
Apologies, I said `.AsString()` but meant [.Str()](https://pkg.go.dev/go.opentelemetry.io/collector/pdata@v1.0.0-rcv0017/pcommon#Value.Str). This is what we currently use for conventions mapping: https://github.com/DataDog/opentelemetry-mapping-go/blob/main/pkg/otlp/attributes/attributes.go#L147. 

To stay consistent, I think we should only add this for value type ValueTypeStr here as well. We can reconsider if there is ask for other value types.